src/tpu.rs

```diff
@@ -3,27 +3,65 @@
 
 use crate::bank::Bank;
 use crate::banking_stage::{BankingStage, BankingStageReturnType};
-use crate::entry::Entry;
+use crate::broadcast_service::BroadcastService;
+use crate::cluster_info::ClusterInfo;
 use crate::fetch_stage::FetchStage;
+use crate::fullnode::TpuRotationSender;
 use crate::poh_service::Config;
 use crate::service::Service;
 use crate::sigverify_stage::SigVerifyStage;
+use crate::tpu_forwarder::TpuForwarder;
 use solana_sdk::hash::Hash;
 use solana_sdk::pubkey::Pubkey;
 use std::net::UdpSocket;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::mpsc::Receiver;
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
 use std::thread;
 
 pub enum TpuReturnType {
     LeaderRotation,
 }
 
-pub struct Tpu {
+pub enum TpuMode {
```

Comment:
Just curious, would having a trait be a better solution? The two implementations of the trait can be Leader and Forwarder, with a defined interface...